impl/src/bin/cargo-raze.rs

```diff
@@ -96,83 +105,85 @@ fn main() -> Result<()> {
+    path_prefix: PathBuf::from(&settings.workspace_path.trim_start_matches("/")),
+    workspace_member_output_dir: settings.workspace_member_dir,
+    vendored_buildfile_name: settings.output_buildfile_suffix,
+    bazel_root: cargo_raze_working_dir,
   };
 
-  let dry_run = options.flag_dryrun.unwrap_or(false);
-  if !dry_run {
-    fs::create_dir_all(&render_details.path_prefix)?;
-  }
-
-  let bazel_file_outputs = match settings.genmode {
+  let bazel_file_outputs = match &settings.genmode {
     GenMode::Vendored => bazel_renderer.render_planned_build(&render_details, &planned_build)?,
     GenMode::Remote => {
-      if !dry_run {
-        let remote_dir = render_details.path_prefix.as_path().join("remote");
-
-        // Clean out the "remote" directory and guarantee that it exists
-        if remote_dir.exists() {
-          let build_glob = format!("{}/BUILD*.bazel", remote_dir.display());
-          for entry in glob::glob(&build_glob)? {
-            if let Ok(path) = entry {
-              fs::remove_file(path)?;
-            }
-          }
-        } else {
-          fs::create_dir_all(&remote_dir)?;
-        }
-
-        if !settings.binary_deps.is_empty() {
-          fs::create_dir_all(
-            render_details
-              .path_prefix
-              .clone()
-              .as_path()
-              .join("lockfiles"),
-          )?;
-        }
-      }
-
       bazel_renderer.render_remote_planned_build(&render_details, &planned_build)?
     }, /* exhaustive, we control the definition */
     // There are no file outputs to produce if `genmode` is Unspecified
     GenMode::Unspecified => Vec::new(),
   };
 
-  for FileOutputs {
-    path,
-    contents,
-  } in bazel_file_outputs
-  {
-    if dry_run {
-      println!("{}:\n{}", path.display(), contents);
-    } else {
-      write_to_file(&path, &contents, options.flag_verbose.unwrap_or(false))?;
+  // Write BUILD files
+  if &settings.genmode == &GenMode::Remote {
+    let remote_dir = render_details
+      .bazel_root
+      .join(render_details.path_prefix)
+      .join("remote");
+
+    // Clean out the "remote" directory so users can easily see what build files are relevant
+    if remote_dir.exists() {
+      let build_glob = format!("{}/BUILD*.bazel", remote_dir.display());
+      for entry in glob::glob(&build_glob)? {
+        if let Ok(path) = entry {
+          fs::remove_file(path)?;
+        }
+      }
     }
   }
 
+  for output in bazel_file_outputs.iter() {
+    if options.flag_dryrun.unwrap_or(false) {
+      println!("{}:\n{}", output.path.display(), output.contents);
+      continue;
+    }
+
+    // Ensure all parent directories exist
+    if let Some(parent) = &output.path.parent() {
+      fs::create_dir_all(parent)?
+    }
+    write_to_file(
+      &output.path,
+      &output.contents,
+      options.flag_verbose.unwrap_or(false),
+    )?;
+  }
+
   Ok(())
 }
 
+/** Helper function for writing rendered files to disk */
```

Comment:
This is a pretty extreme nit, but I'm making the comment because it applies in multiple places.

"Helper function" or "helper struct" in the function or type documentation add very little. Trimming them has more impact:

```
/** Writes rendered files to filesystem. */
```

```
/** Represents the inputs specific to the Remote genmode. */
```

```
/** Gathers inputs for the `genmode = "Remote"` builds. */
```